tests/core/cross/BUILD.bazel

```diff
@@ -33,6 +33,31 @@ go_binary(
     deps = [":platform_lib"],
 )
 
+go_binary(
+    name = "native_bin",
+    srcs = ["main.go"],
+    pure = "on",
+    deps = [":platform_lib"],
+)
+
+go_cross(
+    name = "windows_go_cross",
+    target = ":native_bin",
+    platform = "@io_bazel_rules_go//go/toolchain:windows_amd64", 
+)
+
+go_cross(
+    name = "linux_go_cross",
+    target = ":native_bin",
+    platform = "@io_bazel_rules_go//go/toolchain:linux_amd64", 
+)
+
+go_cross(
+    name = "darwin_go_cross",
+    target = ":native_bin",
+    platform = "@io_bazel_rules_go//go/toolchain:darwin_amd64",
+)
+
```

Comment:
Nit:

This makes sense from a technical implementation perspective. The ability to compose your requirement platforms and use them in `go_cross` is huge.

But from an end-user perspective, it's confusing to read and to use.

Let's say that I setup my targets like this:

```
go_binary(
    name = "native_bin",
    srcs = ["main.go"],
    pure = "on",
    deps = [":platform_lib"],
)

go_cross(
    name = "darwin_go_cross",
    target = ":native_bin",
    platform = "@io_bazel_rules_go//go/toolchain:darwin_amd64",
)
```

which is very typical for normal development nowaday.

What happen when I call `bazel build :all` on a Mac? Would both binaries be identical SHA256 wise?

Compare this to the current setup where we have

```
go_binary(
    name = "linux_bin",
    srcs = ["main.go"],
    goos = "linux",
    deps = [":platform_lib"],
)

go_binary(
    name = "macos_bin",
    srcs = ["main.go"],
    goos = "darwin",
    deps = [":platform_lib"],
)
```

Then I would argue that the current setup is cleaner to read / consume for end-user.

So my question here is: is this the best we could do in term of UX for go_cross?
Perhaps, there is a macro design that could help abstract away go_cross + go_binary into 1 single declaration in a BUILD file? 🤔

Reply:
I would argue that in use cases that have multiple platforms/sdk version combinations its much easier/cleaner to use the `go_cross` setup, especially if the binary has a lot of options eg:
```
go_binary(
  name = "linux_bin",
  goos = "linux",
  srcs = ["main.go"],
  deps = [":lib"],
  gc_goopts = [
    "-N",
    "-l",
    "-more-options-here",
    "-lots-of-options",
    ...
  ],
  gc_linkopts = [
    "-more-opts-in-links",
    "-opt",
    ...
  ],
  data = [
    ":data_dependencies",
    ":more_deps",
    ":makes_the_options_longer",
    ...
  ],
)

go_binary(
  name = "macos_bin",
  srcs = ["main.go"],
  goos = "darwin",
  deps = [":lib"],
  gc_goopts = [
    "-N",
    "-l",
    "-more-options-here",
    "-lots-of-options",
    ...
  ],
  gc_linkopts = [
    "-more-opts-in-links",
    "-opt",
    ...
  ],
  data = [
    ":data_dependencies",
    ":more_deps",
    ":makes_the_options_longer",
    ...
  ],
)

go_binary(
  name = "windows_bin",
  srcs = ["main.go"],
  goos = "windows",
  deps = [":lib"],
  gc_goopts = [
    "-N",
    "-l",
    "-more-options-here",
    "-lots-of-options",
    ...
  ],
  gc_linkopts = [
    "-more-opts-in-links",
    "-opt",
    ...
  ],
  data = [
    ":data_dependencies",
    ":more_deps",
    ":makes_the_options_longer",
    ...
  ],
)
```
versus
```
go_binary(
  name = "native_bin",
  srcs = ["main.go"],
  deps = [":lib"],
  gc_goopts = [
    "-N",
    "-l",
    "-more-options-here",
    "-lots-of-options",
    ...
  ],
  gc_linkopts = [
    "-more-opts-in-links",
    "-opt",
    ...
  ],
  data = [
    ":data_dependencies",
    ":more_deps",
    ":makes_the_options_longer",
    ...
  ],
)

go_cross(
  name = "linux_bin",
  target = ":native_bin",
  platform = "//path_to:my_linux_platform",
)
go_cross(
  name = "macos_bin",
  target = ":native_bin",
  platform = "//path_to:my_macos_platform",
)
go_cross(
  name = "windows_bin",
  target = ":native_bin",
  platform = "//path_to:windows_platform",
)
```

But I will defer to @fmeum

Reply:
I like the design of `go_cross` as it is, but we could think about introducing a `go_cross_binary` macro wrapper that automatically adds `tags = ["manual"]` to the `go_binary`. @sluongng What do you think? This could be done in a follow-up PR though.

Reply:
yup, my thought exactly.

I think `go_cross` is a very nice to have _knob_ for us to build upon.
But it does not comes with a _sane_ default UX that would be needed to make it easier for end users for happy cases.

Im perfectly fine with a macro being introduced in a separate PR 🙏  Thanks!